packages/contracts/contracts/minter-suite/Minters/MinterDAExpV5.sol

```diff
@@ -0,0 +1,583 @@
+            _minimumPriceDecayHalfLifeSeconds
+        );
+    }
+
+    /**
+     * @notice Resets auction details for project `_projectId`, zero-ing out all
+     * relevant auction fields. Not intended to be used in normal auction
+     * operation, but rather only in case of the need to halt an auction.
+     * @param _projectId Project ID to set auction details for.
+     */
+    function resetAuctionDetails(
+        uint256 _projectId,
+        address _coreContract
+    ) external {
+        AuthLib.onlyCoreAdminACL({
+            _coreContract: _coreContract,
+            _sender: msg.sender,
+            _contract: address(this),
+            _selector: this.resetAuctionDetails.selector
+        });
+        DALib.DAProjectConfig
+            storage _auctionProjectConfig = _auctionProjectConfigMapping[
+                _coreContract
+            ][_projectId];
+
+        DALib.resetAuctionDetailsExp(_auctionProjectConfig);
+
+        emit ResetAuctionDetails(_projectId, _coreContract);
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId`.
+     * @param _projectId Project ID to mint a token on.
+     * @param _coreContract Core contract address for the given project.
+     * @return tokenId Token ID of minted token
+     */
+    function purchase(
+        uint256 _projectId,
+        address _coreContract
+    ) external payable returns (uint256 tokenId) {
+        tokenId = purchaseTo(msg.sender, _projectId, _coreContract);
+        return tokenId;
+    }
+
+    // public getter functions
+    /**
+     * @notice Gets the maximum invocations project configuration.
+     * @param _coreContract The address of the core contract.
+     * @param _projectId The ID of the project whose data needs to be fetched.
+     * @return MaxInvocationsLib.MaxInvocationsProjectConfig instance with the
+     * configuration data.
+     */
+    function maxInvocationsProjectConfig(
+        uint256 _projectId,
+        address _coreContract
+    )
+        external
+        view
+        returns (MaxInvocationsLib.MaxInvocationsProjectConfig memory)
+    {
+        return _maxInvocationsProjectConfigMapping[_coreContract][_projectId];
+    }
+
+    /**
+     * @notice Gets the base project configuration.
+     * @param _coreContract The address of the core contract.
+     * @param _projectId The ID of the project whose data needs to be fetched.
+     * @return ProjectConfig instance with the project configuration data.
+     */
+    function projectConfig(
+        uint256 _projectId,
+        address _coreContract
+    ) external view returns (ProjectConfig memory) {
+        return _projectConfigMapping[_coreContract][_projectId];
+    }
+
+    /**
+     * @notice coreContract => projectId => auction parameters
+     */
+    function projectAuctionParameters(
+        uint256 _projectId,
+        address _coreContract
+    ) external view returns (uint256, uint256, uint256, uint256) {
```

Comment:
nit: let's use named args for the return values here, +natspec comments (same comment for the other minters)